examples/python/trigger-pagerduty-incident/handler/handler.py

```diff
@@ -63,94 +69,112 @@ def invoke(self,obj):
         Returns:
             FaaSResponse -- status code and message
         """
-        obj['routing_key'] = self.routingkey
-        obj['event_action'] = self.event_action
         try:
             resp = self.session.post(PAGERDUTY_API_PATH,json=obj)
             resp.raise_for_status()
-            print(resp.text)
+            debug(f'{bgc.OKGREEN}Response Body > {bgc.ENDC}{resp.text}')
             resp_body = json.loads(resp.text)
-            return FaaSResponse('200', 'Successfully {0}ed event action with dedup_key: {1}'.format(self.event_action, resp_body['dedup_key']))
+            return FaaSResponse('200', 'Successfully triggered event action with dedup_key: {0}'.format(resp_body['dedup_key']))
         except requests.HTTPError as err:
             return FaaSResponse('500', 'Could not invoke Pagerduty API > HTTPError: {0}'.format(err))
 
 def handle(req):
     # Validate Event input
     try:
         cevent = json.loads(req)
-    except ValueError as err:
-        res = FaaSResponse('400','invalid JSON > ValueError: {0}'.format(err))
+    except json.decoder.JSONDecodeError as err:
+        res = FaaSResponse('400','Invalid JSON > JSONDecodeError: {0}'.format(err))
         print(json.dumps(vars(res)))
+        debug(f'{bgc.OKBLUE}Event > {bgc.ENDC}{req}')
         return
     
-    # Assert that the Event conforms to CloudEvent Spec
+    # Validate Config file
+    try: 
+        with open(PD_CONFIG, 'r') as pdconfigfile:
+            pdconfig = json.load(pdconfigfile)
+            routingkey=pdconfig['routing_key']
+            event_action=pdconfig['event_action']
+    except OSError as err:
+        res = FaaSResponse('400','Could not read PagerDuty configuration > OSError: {0}'.format(err))
```

Comment:
One could argue that this and the next error should be 500s because they relate to internal server logic and not the provided payload (event data) in this function.

Reply:
I am a little conflicted on treating these functions as a client or server.  I'll have to research the appropriate standard if there is one.

Reply:
Based on the understanding that i have and from research, a 400 error is something that could be fixed by the caller/function user i.e send the function the right event and the correct config and function should work. A 500 error is something that could only be fixed by the function owner or occurs for other reasons that are not under the user's control i.e resource constraints, n/w errors or unexpected exceptions. 

Let me know if you have a different viewpoint.

Reply:
> A 500 error is something that could only be fixed by the function owner

Exactly, that's why I think the error on line 98 should be a `500`.